tests/t0114_gateway_subdomains_test.go

```diff
@@ -13,49 +13,63 @@ import (
 	CIDv1 := "bafkreicysg23kiwv34eg2d7qweipxwosdo2py4ldv42nbauguluen5v6am"
-	// CIDv0 := "QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN"
-	// // CIDv0to1 is necessary because raw-leaves are enabled by default during
-	// // "ipfs add" with CIDv1 and disabled with CIDv0
-	// CIDv0to1 := "bafybeiffndsajwhk3lwjewwdxqntmjm4b5wxaaanokonsggenkbw6slwk4"
-	// CIDv1_TOO_LONG := "bafkrgqhhyivzstcz3hhswshfjgy6ertgmnqeleynhwt4dlfsthi4hn7zgh4uvlsb5xncykzapi3ocd4lzogukir6ksdy6wzrnz6ohnv4aglcs"
-	DIR_CID := "bafybeiht6dtwk3les7vqm6ibpvz6qpohidvlshsfyr7l5mpysdw2vmbbhe" // ./testdirlisting
+	DIR_CID := "bafybeiht6dtwk3les7vqm6ibpvz6qpohidvlshsfyr7l5mpysdw2vmbbhe"
 
 	tests := []CTest{
 		{
-			Name: "request for 127.0.0.1/ipfs/{CID} stays on path",
-			Request: CRequest{
-				Url: fmt.Sprintf("ipfs/%s", CIDv1),
-			},
-			Response: CResponse{
-				StatusCode: 200,
-				Body:       Contains(CID_VAL),
-			},
-		},
-		{
-			Name: "request for localhost/ipfs/{CIDv1} returns HTTP 301 Moved Permanently",
+			Name: "request for {gateway}/ipfs/{CIDv1} returns HTTP 301 Moved Permanently",
 			Request: CRequest{
 				DoNotFollowRedirects: true,
-				Url:                  fmt.Sprintf("/ipfs/%s", CIDv1),
+				Url:                  fmt.Sprintf("%s/ipfs/%s", SubdomainGatewayUrl, CIDv1),
 			},
 			Response: CResponse{
 				StatusCode: 301,
 				Headers: map[string]interface{}{
-					"Location": Contains("http://%s.ipfs.localhost:8080", CIDv1),
+					"Location": Contains("%s://%s.ipfs.%s", SubdomainGatewayScheme, CIDv1, SubdomainGatewayHost),
 				},
 			},
 		},
+		{
+			Name: "request for {gateway}/ipfs/{CIDv1} returns HTTP 301 Moved Permanently (sugar)",
+			Request: Request().
+				URL("http://example.com/ipfs/%s", CIDv1).
+				DoNotFollowRedirects().
+				Request(),
+			Response: Expect().
+				Status(301).
+				Headers(
+					Header("Location").
+						// TODO: this works only because we use example.com in our tests.
+						// It should be:
+						// Contains("%s://%s.ipfs.%s", SubdomainGatewayScheme, CIDv1, SubdomainGatewayHost)
+						// I am trying to avoid this syntax.
+						// The other option is to force the tested gateway to use example.com.
+						Contains("http://%s.ipfs.example.com", CIDv1),
+				).
+				Response(),
+		},
 		{
 			Name: "request for {cid}.ipfs.localhost/api returns data if present on the content root",
 			Request: CRequest{
-				RawURL: fmt.Sprintf("http://%s.ipfs.localhost:8080/api/file.txt", DIR_CID),
+				Url: fmt.Sprintf("%s://%s.ipfs.%s/api/file.txt", SubdomainGatewayScheme, DIR_CID, SubdomainGatewayHost),
 			},
 			Response: CResponse{
 				Body: Contains("I am a txt file"),
 			},
 		},
+		{
+			Name: "request for {cid}.ipfs.localhost/api returns data if present on the content root (sugar)",
+			Request: Request().
+				URL("http://%s.ipfs.example.com/api/file.txt", DIR_CID).
+				Request(),
+			Response: Expect().
+				Status(200).
+				Body("I am a txt file\n").
+				Response(),
+		},
```

Comment:
Note the test above, without sugar, we have to use URL:
`Url: fmt.Sprintf("%s://%s.ipfs.%s/api/file.txt", SubdomainGatewayScheme, DIR_CID, SubdomainGatewayHost),`

the sugar rewrites the URLs for you, which means a test implementer can use
`URL("http://%s.ipfs.example.com/api/file.txt", DIR_CID).`
and the testing tool rewrites it.